source/mixins/versionable.js

```diff
@@ -0,0 +1,19 @@
+Space.messaging.Versionable = {
+
+  schemaVersion: 1,
+
+  onConstruction(data) {
+    if (_.isObject(data)) this._migrateToLatestVersion(data);
```

Comment:
it's also really a transformation rather than migration since we're not changing data.

Maybe put the version check in the `onConstruction`, then call a `_transformLegacySchema` method if needed